File: 2024 czerwiec/INF.04-01-24.06-SG/Aplikacja Konsolowa/C++/main.cpp
#include <iostream>
#include <random>

using namespace std;

class DiceThrow
{
public:
    int randomNumbers[10];

    void displayRandomNumbers(int numberOfDice)
    {
        for (int i = 0; i < numberOfDice; i++)
        {
            cout << "Kostka " << i + 1 << ": " << randomNumbers[i] << endl;
        }
    }

    void generateRandomNumbers(int numberOfDice)
    {
        random_device rd;
        mt19937 gen(rd());
        uniform_int_distribution<> distr(1, 6);

        for (int i = 0; i < numberOfDice; i++)
        {
            randomNumbers[i] = distr(gen);
        }
    }

    int calculatePoints(int numberOfDice)
    {
        int points = 0;
        bool isRepeating = false;

        for (int i = 0; i < numberOfDice; i++)
        {
            for (int j = 0; j < numberOfDice; j++)
            {
                if (randomNumbers[i] == randomNumbers[j] && i != j)
                {
                    points += randomNumbers[j];
                    isRepeating = true;
                    randomNumbers[j] = 0;
                }
            }
            if (isRepeating == true)
            {
                points += randomNumbers[i];
                isRepeating = false;
            }
        }
        return points;
    }
};

int main()
{
    int numberOfDices;

    do
    {
        cout << "Ile kostek chcesz rzucić? (3 - 10)" << endl;
        cin >> numberOfDices;
    } while (numberOfDices < 3 || numberOfDices > 10);

    DiceThrow newDiceThrow;

    newDiceThrow.generateRandomNumbers(numberOfDices);
    newDiceThrow.displayRandomNumbers(numberOfDices);
    cout << "Liczba uzyskanych punktów: " << newDiceThrow.calculatePoints(numberOfDices) << endl;

    string userInput = "t";

    while (userInput != "t" || userInput != "n")
    {
        cout << "Jeszcze raz? (t/n)" << endl;
        cin >> userInput;
        if (userInput == "t")
        {
            newDiceThrow.generateRandomNumbers(numberOfDices);
            newDiceThrow.displayRandomNumbers(numberOfDices);
            cout << "Liczba uzyskanych punktów: " << newDiceThrow.calculatePoints(numberOfDices) << endl;
        }
        else
        {
            exit(0);
        }
    }

    return 0;
}

File: 2024 czerwiec/INF.04-01-24.06-SG/Aplikacja Konsolowa/Python/main.py
import random

class DiceThrow:
    def __init__(self):
        self.randomNumbers = {}
    
    def displayRandomNumbers(self, numberOfDice):
        for i in range(numberOfDice):
            print("Kostka", i + 1, ":", self.randomNumbers[i])
    
    def generateRandomNumbers(self, numberOfDice):
        for i in range(numberOfDice):
            self.randomNumbers[i] = random.randint(1, 6)
            
    def calculatePoints(self, numberOfDice):
        points = 0
        isRepeating = False
        
        for i in range(numberOfDice):
            for j in range(numberOfDice):
                if self.randomNumbers[i] == self.randomNumbers[j] and i != j:
                    points += self.randomNumbers[j]
                    isRepeating = True
                    self.randomNumbers[j] = 0
            if isRepeating == True:
                points += self.randomNumbers[i]
                isRepeating = False
        
        return points
    
while True:
    numberOfDices = int(input("Ile kostek chcesz rzucić? (3-10)\n"))
    if numberOfDices >= 3 or numberOfDices <= 10:
        break

newDiceThrow = DiceThrow()

newDiceThrow.generateRandomNumbers(numberOfDices)
newDiceThrow.displayRandomNumbers(numberOfDices)
print("Liczba uzyskanych punktów:", newDiceThrow.calculatePoints(numberOfDices))
    
userInput = "t"

while userInput != "t" or userInput != "n":
    userInput = input("Jeszcze raz (t/n)\n")
    
    if userInput == "t":
        newDiceThrow.generateRandomNumbers(numberOfDices)
        newDiceThrow.displayRandomNumbers(numberOfDices)
        print("Liczba uzyskanych punktów:", newDiceThrow.calculatePoints(numberOfDices))
    else:
        exit()